File: Python/ChatGPT_3.5_Free/Singleton/round2/test_ChatGPT_R2_Singleton.py
import pytest
from ChatGPT_R2_Singleton import Singleton

def test_singleton_instance_creation():
    # Arrange
    instance1 = Singleton()
    instance2 = Singleton()

    # Act & Assert
    assert instance1 is instance2  # Both should be the same instance

def test_singleton_value_setting():
    # Arrange
    instance = Singleton()
    value = 10

    # Act
    instance.set_value(value)

    # Assert
    assert instance.get_value() == value

def test_singleton_reset():
    # Arrange
    instance1 = Singleton()
    instance1.set_value(20)

    # Act
    instance1.reset()
    instance2 = Singleton()

    # Assert
    assert instance1 is not instance2  # New instance after reset
    assert instance2.get_value() is None  # Ensure the new instance has reset value

File: Java/src/test/java/com/java/ChatGPTFree/Singleton/round2/test_ChatGPT_R2_Singleton.java
package com.java.ChatGPTFree.Singleton.round2;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class DatabaseConnectionTest {

    @Test
    void testGetInstance() {
        // Arrange
        ChatGPT_R2_Singleton firstInstance;

        // Act
        firstInstance = ChatGPT_R2_Singleton.getInstance();
        ChatGPT_R2_Singleton secondInstance = ChatGPT_R2_Singleton.getInstance();

        // Assert
        assertNotNull(firstInstance); // Ensure the instance is not null
        assertSame(firstInstance, secondInstance); // Ensure both references point to the same instance
    }

    @Test
    void testGetConnectionStatus() {
        // Arrange
        ChatGPT_R2_Singleton dbConnection = ChatGPT_R2_Singleton.getInstance();

        // Act
        String status = dbConnection.getConnectionStatus();

        // Assert
        assertEquals("Database connection established.", status); // Check the connection status
    }
}
